### DISCUSSION TITLE
How to make the example plots from the mesh objects?

### DISCUSSION THREAD

**Discussion by sugatoray** [nschloe/pygmsh#453] [General]
Please share some example code (using python) on making the figures similar to the ones shared in the repo-README file. What I am looking for is, given you have a mesh object in pygmsh, how do I create a figure?

**Comment by nschloe** [comment]
All the code is in the readme, right next to the figures.

**Reply by sugatoray** [reply]
Do you mean this? Also, I am trying to use pygmsh in Google Colab notebook environment. Does the output generated happens in a new window or can be placed inline in the notebook? 

![image](https://user-images.githubusercontent.com/10201242/116936234-26d60c80-ac2d-11eb-848a-daa5cb3296b6.png)

**Reply by nschloe** [reply]
pygmsh doesn't have graphical output.

**Reply by akbarpn136** [reply]
Hope that this is the right place for me to ask about. Any example of how to plot a mesh with matplotlib?

**Reply by SunilAnandatheertha** [reply]
You could write the output to a .vtk file using meshio and then read it using say pyvista and visualize your mesh.

**Reply by akbarpn136** [reply]
> You could write the output to a .vtk file using meshio and then read it using say pyvista and visualize your mesh.

Thank you @SunilAnandatheertha  for the reply.

Is it possible to utilize mesh from `mesh = geom.generate_mesh()` let say:
`mesh.points` or `mesh.cells` or maybe `mesh.faces` into pyvista without writing to other mesh format first? For example here

```python
import pyvista as pv

vertices = mesh.points
faces = mesh.faces

surf = pv.PolyData(vertices, faces)
surf.plot()
```

Thank you for your guidance, I'm new to pyvista too
